### DISCUSSION TITLE
Is there anyway in GLSP , we can show properties for a  box in diagram ?

### DISCUSSION THREAD

**Discussion by midhun8454** [eclipse-glsp/glsp#439] [Ideas]
What exactly i mean is when i click on any boxes with in diagram , may be it might be having different properties and objects associated with it ? So like when you click on any one box , it should show modal or a right side box like palette , briefing all the properties and objects associated with it.
Not sure if that feature is already there or not , tried , but didnt get any apt one.

Kind of self explanatory for each box , am not asking about typehints

**Comment by planger** [comment]
This is typically done with [popups](https://github.com/eclipse-glsp/glsp-server/blob/c96935795bedc798dfc83c82ca46dc5a50860475/examples/org.eclipse.glsp.example.workflow/src/org/eclipse/glsp/example/workflow/WorkflowDiagramModule.java#L131) or you create a [custom UI extension](https://github.com/eclipse-glsp/glsp-client/blob/756081858c5ca56f8cae94fa5601dcd3c3beab2a/examples/workflow-glsp/src/direct-task-editing/di.config.ts#L23) or external widget, such as a properties view, that queries additional information from the selected element.

**Reply by midhun8454** [reply]
okay , any handy example or screenshot to substantiate same ?

**Reply by planger** [reply]
Here is a bug report with a screencast where the popup is visible (the bug is now fixed, but its screencast should still serve this purpose ;-)):
https://github.com/eclipse-glsp/glsp/issues/184

Of course you can style the popups using CSS to your needs. Admittedly the one in the workflow example is not very pretty.

I added links to code in my answer above.

**Reply by midhun8454** [reply]
got it. Again the data inside the popup can be customisable , right ? Like a tree structure or accordion like ?

**Reply by planger** [reply]
Right, you basically provide HTML for the content of the popup:
https://github.com/eclipse-glsp/glsp-server/blob/c96935795bedc798dfc83c82ca46dc5a50860475/examples/org.eclipse.glsp.example.workflow/src/org/eclipse/glsp/example/workflow/WorkflowPopupFactory.java#L46

**Reply by midhun8454** [reply]
make sense

**Reply by mshahzadiftikhar** [reply]
is there a way to show these properties when VSCode is GLSP client?

**Reply by ndoschek** [reply]
Hi @mshahzadiftikhar,

At the moment, there isn't a ready-made solution for showing properties when VS Code is used as a GLSP client.
However, you could follow a similar approach as mentioned in the thread above.
You can either make use of the [`PopupModelFactory`](https://github.com/eclipse-glsp/glsp-server-node/blob/main/packages/server/src/common/features/popup/popup-model-factory.ts) (I've updated the link to the current default node server implementation) or implement a custom UI extension as described in [this UI extension guide from the GLSP documentation](https://eclipse.dev/glsp/documentation/ui-extensions/).

Hope this helps!

Cheers,
Nina

**Comment by ndoschek** [comment]
If you will be using GLSP inside a Theia application, you can also use their [property view](https://github.com/ndoschek/theia/blob/8a3ec7de011a88c0a17ee71bcee3e6ff26c0729f/packages/property-view/README.md). You can also have a look at [this blog post](https://eclipsesource.com/blogs/2021/03/10/a-property-view-for-eclipse-theia/).

**Reply by midhun8454** [reply]
cool. Any snapshot on same will be awesome .
Also is this something comes when you click on any single node in a diagram ? @ndoschek

**Reply by midhun8454** [reply]
am not sure , i guess this is not what am asking .. My ask is when you click any node in diagram , i need to have property view coming as popup or in right pane @ndoschek

**Reply by ndoschek** [reply]
Yes, that's correct - the property view is not a popup, it is a separate widget inside the Theia editor, as shown in this example:

![ecore-glsp-property-view](https://user-images.githubusercontent.com/47414588/139252384-652771fe-9962-492b-a852-30c2775982b2.png)

**Reply by midhun8454** [reply]
okay , thats something am not looking :(

**Reply by CamilleLetavernier** [reply]
> i need to have property view coming as popup or in right pane

Since the properties view is a standard Theia view, you can dock it wherever you like - including the right side of the diagram :) But it's indeed not a popup
